In [1]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn import metrics
from sklearn.tree import DecisionTreeClassifier

import numpy as np
import seaborn as sns #
import matplotlib.pyplot as plt

In [2]:
train = pd.read_csv('data2/train.csv', index_col = 'ID')
test = pd.read_csv('data2/test.csv', index_col = 'ID')

In [3]:
print(train.shape)
print(test.shape)

(699, 16)
(430, 15)


In [4]:
train.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 699 entries, 1 to 699
Data columns (total 16 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   stag          516 non-null    float64
 1   event         699 non-null    int64  
 2   gender        699 non-null    object 
 3   age           699 non-null    float64
 4   industry      672 non-null    object 
 5   profession    699 non-null    object 
 6   traffic       699 non-null    object 
 7   coach         699 non-null    object 
 8   head_gender   699 non-null    object 
 9   greywage      679 non-null    object 
 10  way           678 non-null    object 
 11  extraversion  699 non-null    float64
 12  independ      699 non-null    float64
 13  selfcontrol   666 non-null    float64
 14  anxiety       699 non-null    float64
 15  novator       699 non-null    float64
dtypes: float64(7), int64(1), object(8)
memory usage: 92.8+ KB


In [5]:
test.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 430 entries, 700 to 1129
Data columns (total 15 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   stag          321 non-null    float64
 1   gender        430 non-null    object 
 2   age           430 non-null    float64
 3   industry      430 non-null    object 
 4   profession    430 non-null    object 
 5   traffic       430 non-null    object 
 6   coach         430 non-null    object 
 7   head_gender   360 non-null    object 
 8   greywage      430 non-null    object 
 9   way           430 non-null    object 
 10  extraversion  430 non-null    float64
 11  independ      430 non-null    float64
 12  selfcontrol   385 non-null    float64
 13  anxiety       430 non-null    float64
 14  novator       430 non-null    float64
dtypes: float64(7), object(8)
memory usage: 53.8+ KB


- 총 데이터 수 699개 (train)
- 결측치 채워야 하는것 (train)
- stag : 근무기간(월) 183개
- industry : 직원의 산업 27개
- greywage : 임금투명성 20개
- way : 출퇴근 방법 21개
- selfcontrol : 자제력 33개


--

- 총 데이터 수 430개 (test)
- 결측치 채워야 하는것 (test)
- stag : 근무기간(월) 109개
- head_gender : 지도자 성별 70개
- selfcontrol : 자제력 45개

In [10]:
train.corr()

stag     event       age  extraversion  independ  \
stag          1.000000 -0.070421 -0.214415     -0.078248 -0.013056   
event        -0.070421  1.000000  0.008723      0.013961 -0.007915   
age          -0.214415  0.008723  1.000000     -0.169429  0.074582   
extraversion -0.078248  0.013961 -0.169429      1.000000 -0.208904   
independ     -0.013056 -0.007915  0.074582     -0.208904  1.000000   
selfcontrol   0.102294 -0.049588  0.032276     -0.512422 -0.202594   
anxiety       0.042807 -0.022668  0.060051     -0.150632 -0.418896   
novator      -0.039662  0.022554  0.020334      0.287073  0.038421   

              selfcontrol   anxiety   novator  
stag             0.102294  0.042807 -0.039662  
event           -0.049588 -0.022668  0.022554  
age              0.032276  0.060051  0.020334  
extraversion    -0.512422 -0.150632  0.287073  
independ        -0.202594 -0.418896  0.038421  
selfcontrol      1.000000 -0.074452 -0.573815  
anxiety         -0.074452  1.000000  0.255866  
novator         -0.573815  0.255866  1.000000

In [7]:
pt1 = train.pivot_table(values ='age', index = ['event', 'gender'], aggfunc='mean')

In [8]:
pt1

age
event gender           
0     f       30.205915
      m       31.285714
1     f       30.326863
      m       31.144231

In [9]:
# df = raw.

SyntaxError: invalid syntax (<ipython-input-9-bcfd42e750ae>, line 1)

In [ ]:
sns.clustermap(train.corr(),
              annot = True, # 실제값 화면에 나타내기
            cmap = 'RdYlBu_r',# Red, Yellow, Blue 색상으로 표시
               vmin = -1, vmax = 1, #컬러차트 -1 ~1 범위로 표시
              )